pkg/api/query/v1.go

```diff
@@ -338,7 +338,8 @@ func (qapi *QueryAPI) query(r *http.Request) (interface{}, []error, *api.ApiErro
 	span, ctx := tracing.StartSpan(ctx, "promql_instant_query")
 	defer span.Finish()
 
-	qry, err := qe.NewInstantQuery(qapi.queryableCreate(enableDedup, replicaLabels, storeDebugMatchers, maxSourceResolution, enablePartialResponse, false), r.FormValue("query"), ts)
+	tenant := r.Header.Get("THANOS-TENANT")
```

Comment:
`X-` prefix also?

Reply:
Yes, let's make the prefix as well configurable to be consistent with the receiver.